src/index.js

```diff
@@ -52,7 +57,7 @@ function parseValue(value) {
   } else if (_.isPlainObject(value)) {
     return parseMap(value);
   } else {
-    return value;
+    return parseEndValue(value);
```

Comment:
`parseEndValue` is the worst name in the world. I should rename it to `parseValue` or maybe you have a better proposal :)